inc/lib/neopixel.h

```diff
@@ -0,0 +1,133 @@
+/*
+
+Original source of this neopixel driver: https://github.com/lavallc/nrf51-neopixel
+
+The MIT License (MIT)
+
+Copyright (c) 2013 Lava
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of
+this software and associated documentation files (the "Software"), to deal in
+the Software without restriction, including without limitation the rights to
+use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+the Software, and to permit persons to whom the Software is furnished to do so,
+subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+*/
+
+ #ifndef NEOPIXEL_H
+ #define NEOPIXEL_H
+
+
+#include <stdbool.h>
+#include <stdint.h>
+#include "nrf_delay.h"
+#include "nrf_gpio.h"
+
+
+//These defines are timed specific to a series of if statements and will need to be changed
+//to compensate for different writing algorithms than the one in neopixel.c
+#define NEOPIXEL_SEND_ONE	NRF_GPIO->OUTSET = (1UL << PIN); \
```

Comment:
I want to minimise changes to the neopixel driver because it's from an upstream project (see comment at top of .h file).  You can see in the commits that I added the pristine source files, then made baby-step modifications, just enough so that it would work for us.  No need to make unnecessary changes for code style.  It works as it is :)